pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -97,30 +120,63 @@ func (rc *remoteClient) setConfig(watchCtx context.Context, kubeconfig []byte) e
 	}
 	rc.client = remoteClient
 
-	newWatcher, err := rc.client.Watch(watchCtx, &kueue.WorkloadList{})
+	err = rc.startWatcher(watchCtx, kueue.GroupVersion.WithKind("Workload").GroupKind().String(), &workloadKueueWatcher{})
```

Comment:
Why are we watching workloads, and not Jobs directly? If we are only watching workloads, how do we get status updates on the remote Job status update?

Reply:
Hm, here: `err := rc.startWatcher(watchCtx, kind, watcher, client.HasLabels{constants.PrebuiltWorkloadLabel})`?

Why do we then actually watch both the workload and job? I would imagine we watch the job and update the local status. It is not clear to me why we watch both. Please elaborate / add comment.

Reply:
So, to confirm IIUC we watch jobs for the live status updates.

Still, we also watch remote workloads to trigger local workload reconciliation. This is needed for example when the workload is finished.

Reply:
yes